pkg/storage/bloom/v1/ast_extractor_test.go

```diff
@@ -0,0 +1,105 @@
+package v1_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/loki/v3/pkg/logql/syntax"
+	v1 "github.com/grafana/loki/v3/pkg/storage/bloom/v1"
+)
+
+func TestExtractLabelMatchers(t *testing.T) {
+	tt := []struct {
+		name   string
+		input  string
+		expect []v1.LabelMatcher
+	}{
+		{
+			name:  "basic label matcher",
+			input: `{app="foo"} | key="value"`,
+			expect: []v1.LabelMatcher{
+				v1.PlainLabelMatcher{Key: "key", Value: "value"},
+			},
+		},
+
+		{
+			name:  "or label matcher",
+			input: `{app="foo"} | key1="value1" or key2="value2"`,
+			expect: []v1.LabelMatcher{
+				v1.OrLabelMatcher{
+					Left:  v1.PlainLabelMatcher{Key: "key1", Value: "value1"},
+					Right: v1.PlainLabelMatcher{Key: "key2", Value: "value2"},
+				},
+			},
+		},
+
+		{
+			name:  "and label matcher",
+			input: `{app="foo"} | key1="value1" and key2="value2"`,
+			expect: []v1.LabelMatcher{
+				v1.AndLabelMatcher{
+					Left:  v1.PlainLabelMatcher{Key: "key1", Value: "value1"},
+					Right: v1.PlainLabelMatcher{Key: "key2", Value: "value2"},
+				},
+			},
+		},
+
+		{
+			name:  "multiple label matchers",
+			input: `{app="foo"} | key1="value1" | key2="value2"`,
+			expect: []v1.LabelMatcher{
+				v1.PlainLabelMatcher{Key: "key1", Value: "value1"},
+				v1.PlainLabelMatcher{Key: "key2", Value: "value2"},
+			},
+		},
+
+		{
+			name:  "unsupported label matchers",
+			input: `{app="foo"} | key1=~"value1"`,
```

Comment:
nit: later on, we should be able to support these kind of regexes.
```
key=~"value" --> key="value"
key=~"value|other" --> key="value" or key="other"
key=".+" --> we can just test that the series contain the label (`key` in this case)
```

We do this already for the filter itself in the logQL engine:
https://github.com/grafana/loki/blob/main/pkg/logql/log/filter.go#L639-L667

We started supporting this for blooms, but disabled it because we found a bug that was likely caused by this optimization and left it disabled. Here's the original PR adding it:  https://github.com/grafana/loki/pull/12096